backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/audit/data/repository/DataExportAuditLogRepository.java

```diff
@@ -0,0 +1,101 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.audit.data.repository;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import de.metas.audit.data.Action;
+import de.metas.audit.data.ExternalSystemParentConfigId;
+import de.metas.audit.data.model.DataExportAuditId;
+import de.metas.audit.data.model.DataExportAuditLog;
+import de.metas.audit.data.model.DataExportAuditLogId;
+import de.metas.process.PInstanceId;
+import de.metas.util.NumberUtils;
+import de.metas.util.Services;
+import lombok.NonNull;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.compiere.model.I_Data_Export_Audit_Log;
+import org.springframework.stereotype.Repository;
+
+import java.util.Objects;
+
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+@Repository
+public class DataExportAuditLogRepository
+{
+	private final IQueryBL queryBL = Services.get(IQueryBL.class);
+
+	@NonNull
+	public DataExportAuditLog save(@NonNull final DataExportAuditLog logEntry)
+	{
+		final I_Data_Export_Audit_Log record = InterfaceWrapperHelper.newInstance(I_Data_Export_Audit_Log.class);
+
+		record.setData_Export_Audit_ID(logEntry.getDataExportAuditId().getRepoId());
+		record.setData_Export_Action(logEntry.getAction().getCode());
+		record.setExternalSystem_Config_ID(NumberUtils.asInteger(logEntry.getExternalSystemConfigId(), -1));
+		record.setAD_PInstance_ID(NumberUtils.asInteger(logEntry.getAdPInstanceId(), -1));
+
+		saveRecord(record);
+
+		return recordToDataExportAuditLog(record);
+	}
+
+	@NonNull
+	public ImmutableSet<Integer> getExternalSystemConfigIds(@NonNull final DataExportAuditId dataExportAuditId)
+	{
+		return queryBL.createQueryBuilder(I_Data_Export_Audit_Log.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_Data_Export_Audit_Log.COLUMNNAME_Data_Export_Audit_ID, dataExportAuditId.getRepoId())
+				.create()
+				.listDistinct(I_Data_Export_Audit_Log.COLUMNNAME_ExternalSystem_Config_ID, Integer.class)
+				.stream()
+				.filter(Objects::nonNull)
+				.collect(ImmutableSet.toImmutableSet());
+	}
+
+	@NonNull
+	public ImmutableList<DataExportAuditLog> getByDataExportAuditId(@NonNull final DataExportAuditId dataExportAuditId)
+	{
+		return queryBL.createQueryBuilder(I_Data_Export_Audit_Log.class)
+				.addOnlyActiveRecordsFilter()
+				.addEqualsFilter(I_Data_Export_Audit_Log.COLUMNNAME_Data_Export_Audit_ID, dataExportAuditId.getRepoId())
+				.create()
+				.stream()
+				.map(this::recordToDataExportAuditLog)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	@NonNull
+	private DataExportAuditLog recordToDataExportAuditLog(@NonNull final I_Data_Export_Audit_Log record)
```

Comment:
consider making it static

Reply:
... and usually, at metasfresh, we are naming these methods as `toDataExportAuditLog`

Reply:
made static & renamed